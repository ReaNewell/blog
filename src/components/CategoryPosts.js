import React from 'react';
import SummarizedPost from './SummarizedPost';

class CategoryPosts extends React.Component {
    render() {
        return (
            <div className="recent-posts">
                <h2 className="recent-posts__title">{`Articles Related To ${this.props.category}`}</h2>
                {this.props.posts.map((post) => (
                    <SummarizedPost 
                        key = {post.id}
                        title = {post.title}
                        imageName = {post.postPictureName}
                        imagePath = {post.postPicture}
                        date = {post.postDate}
                    />
                ))}
            </div>
        )
    }
}

export default CategoryPosts;