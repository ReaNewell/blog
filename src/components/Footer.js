import React from 'react';

export const Footer = () => (
    <div className="footer">
        <img alt='site-logo' src='/images/rea-logo.png'/>
    </div>
);