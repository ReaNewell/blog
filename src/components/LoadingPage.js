import React from 'react';

export const LoadingPage = () => (
    <div className='loading-page'>
        <div className='loading-page__logo'/>
    </div>
);