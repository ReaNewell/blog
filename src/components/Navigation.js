import React from 'react';
import { Link } from 'react-router-dom';

export const Navigation = () => (
    <div>
        <Link to="/">Home</Link>
        <Link to="/">About</Link>
        <Link to="/">Contact</Link>
        <Link to="/">Portfolio</Link>
    </div>
);