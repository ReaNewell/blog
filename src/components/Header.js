import React from 'react';
import { Navigation } from './Navigation';

export const Header = () => (
    <div>
        <Navigation />
    </div>
);