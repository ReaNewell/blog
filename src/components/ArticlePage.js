import React from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import { Footer } from './Footer';
import { Header } from './Header';
import SummarizedPost from './SummarizedPost';
import SideBar from './SideBar';
import moreArticles from '../selectors/moreArticles';
import dateFormatter from '../tools/dateFormatter';

export class ArticlePage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            html: this.props.post.body.replace(/^"(.*)"$/, '$1')
        }
    } 
    componentDidMount() {
        window.scroll(0, 0);

        // Selects this the body element of this component and assign the HTML value of the post to that element.
        const body = ReactDOM.findDOMNode(this).getElementsByClassName('article__body')[0];
        body.innerHTML = this.state.html;
    }
    componentDidUpdate() {
        window.scroll(0, 0);
    }
    render() {
        return (
            <div className="article-page">
                <Header />
                    <article className="article">
                        <h2 className="article__title">{this.props.post.title.toUpperCase()}</h2>
                        <p className="article__date">{dateFormatter(this.props.post.postDate.day, this.props.post.postDate.month, this.props.post.postDate.year)}</p>
                        {this.props.post.postPicture && <img className="article__image" src={this.props.post.postPicture} alt={this.props.post.postPictureName}/>}
                        <div className="article__body" />

                        <div className='share-buttons'>
                            <a className="share-buttons__share-button" href={`http://facebook.com/sharer.php?u=https%3A%2F%2Fblog.reanewell.com/posts/${this.props.post.title.replace(/ /g, "-")}`}>Facebook</a>
                            <a className="share-buttons__share-button" href={`https://twitter.com/intent/tweet?url=https%3A%2F%2Fblog.reanewell.com/${this.props.post.title.replace(/ /g, "-")}`}>Twitter</a>
                            <a className="share-buttons__share-button" href={`https://www.linkedin.com/shareArticle?mini=true&url=https%3A%2F%2Fblog.reanewell.com/${this.props.post.title.replace(/ /g, "-")}&title=${this.props.post.title}`}>LinkedIn</a>
                        </div>

                        <h2 className='article__header'>More Articles</h2>
                        {this.props.moreArticles.map((post) => (
                            <SummarizedPost 
                                key = {post.id}
                                title = {post.title}
                                imageName = {post.postPictureName}
                                imagePath = {post.postPicture}
                                date = {post.postDate}
                            />
                        ))}
                    </article>
                <SideBar />
                <Footer />
            </div>
        )
    }
}
const mapStateToProps = (state, props) => {
    const post = state.posts.find((post) => post.title === props.match.params.title.replace(/-/g, ' '));

    return ({
        post,
        moreArticles: moreArticles(state.posts, post.id, post.category)
    })
};

export default connect(mapStateToProps)(ArticlePage);