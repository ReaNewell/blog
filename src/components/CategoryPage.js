import React from 'react';
import { connect } from 'react-redux';
import categoryFilter from '../selectors/categoryFilter';
import { Footer } from './Footer';
import { Header } from './Header';

import SideBar from './SideBar';
import CategoryPosts from './CategoryPosts';

export const CategoryPage = (props) => {
    componentDidMount = () => {
        window.scroll(0, 0);
    }
    componentDidUpdate = () => {
        window.scroll(0, 0);
    }
    return(
        <div className="homepage">
            <Header />
            <CategoryPosts 
                category={props.category}
                posts={props.posts}
            />
            <SideBar />
            <Footer />
        </div>
    )
};
const mapStateToProps = (state, props) => ({
    category: props.match.params.category,
    posts: categoryFilter(state.posts, props.match.params.category)
});

export default connect(mapStateToProps)(CategoryPage);