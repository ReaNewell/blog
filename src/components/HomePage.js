import React from 'react';

import { Footer } from './Footer';
import { Header } from './Header';
import RecentPosts from './RecentPosts';
import SideBar from './SideBar';

const HomePage = () => {
    window.scroll(0, 0);
    return (
        <div className="homepage">
            <Header />
            <RecentPosts />
            <SideBar />
            <Footer />
        </div>
    )
};

export default HomePage;