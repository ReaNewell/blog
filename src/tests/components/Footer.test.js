import React from 'react';
import { shallow } from 'enzyme';
import { Footer } from '../../components/Footer';

test('should render Footer component correctly', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper).toMatchSnapshot();
});