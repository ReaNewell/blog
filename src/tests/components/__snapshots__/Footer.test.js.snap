// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component correctly 1`] = `
<div
  className="footer"
>
  <img
    alt="site-logo"
    src="/images/rea-logo.png"
  />
</div>
`;
